### ISSUE TITLE
Generate an url builder method for every resource path

### ISSUE THREAD

**Issue by aleksandr-vin** [go-swagger/go-swagger#119]
Started from the need of smth. like [flask.url_for in Python](http://flask.pocoo.org/docs/0.10/api/#flask.url_for) to return a newly created resource url in Location header.

Considering that _/generator/templates/server/operation.gotmpl_ is a nice place for such url builder function.
## Below are my suggestions

At first point I think of extending the point to get the ability to build urls for any **unique** resource path mentioned in spec.

For ex. all these 5 operations
- _GET /events_
- _POST /events_
- _GET /events/{id}_
- _PUT /events/{id}_
- _DELETE /events/{id}_

Will need only two url builder functions, smth. like:
- `events.BuildUrl()`
- `events.BuildUrl(Id)`

But this case seems to have a leak in abstraction between operation and its url. If we decide to replace the _PUT /events/{id}_ **operation** to another url keeping the _operationId_ unchanged -- we will need to update the call to its url builder function. But actually we have not changed the semantic of the **operations** so we need a concrete url builder call for it.

In other words I think we need a one-to-one url-builder functions for each **operation** to be able to use the proper one based on the operation semantics.

**Comment by casualjim** [comment]
this was implemented in the 0.7.x version
